src/socketio.quiver.js

```diff
@@ -855,22 +932,57 @@ QuiverSocialProvider.prototype.changeRoster = function(userId, clientSuffix, inv
 };
 
 /**
- * Interpret messages from the server to this as owner.
+ * @private
+ * @param {!QuiverSocialProvider.server_} server The server through which the message was delivered
+ * @param {Object} msg Encrypted message from the server
+ **/
+QuiverSocialProvider.prototype.onEncryptedMessage_ = function(server, msg) {
+  if (!msg) {
+    return;
+  }
+
+  if (msg.cmd === 'ping') {  // TODO: Better name than ping for a broadcast
+    // Only pings are allowed to go unencrypted.
+    // Reply to the ping with an encrypted intro msg.
+    var friendDesc = this.configuration_.friends[msg.from];
+    if (friendDesc) {
+      var serverKey = QuiverSocialProvider.serverKey_(server);
+      var introMsg = this.makeIntroMsg_(friendDesc);
+      this.emitEncrypted_(this.connections_[serverKey].socket, msg.from,
+          introMsg);
+    }
```

Comment:
might be good to have console.warn if we get a ping from someone not in our friends list